src/scripts/main.js

```diff
@@ -1 +1,54 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+firstPromise
+  .then((res) => createMessage(res, 'success'))
+  .catch((err) => createMessage(err, 'error'));
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', (e) => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+
+    resolve('Second promise was resolved');
+  });
+});
+
+secondPromise.then((res) => createMessage(res, 'success'));
+
+const leftClick = new Promise((resolve) => {
+  document.addEventListener('click', () => resolve('success'));
+});
+
+const rightClick = new Promise((resolve) => {
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    resolve('success');
+  });
+});
+
+const thirdPromise = Promise.all([leftClick, rightClick]);
```

Comment:
The `thirdPromise` is created using `Promise.all([leftClick, rightClick])`. This is correct as it waits for both promises to resolve. Ensure that both `leftClick` and `rightClick` are resolved by their respective events before this promise resolves.